calicoctl/calico_ctl/pool.py

```diff
@@ -188,10 +192,38 @@ def ip_pool_remove(cidrs, version):
     :return: None
     """
     for cidr in cidrs:
+        # Get the existing IP Pool so that we can disable it,
         try:
-            client.remove_ip_pool(version, IPNetwork(cidr))
+            pool = client.get_ip_pool_config(version, IPNetwork(cidr))
         except KeyError:
             print "%s is not a configured pool." % cidr
+            sys.exit(1)
+
+        try:
+            # Disable the pool to prevent any further allocation blocks from
+            # being assigned from the pool.  Existing allocation blocks will
+            # still exist and may be allocated from until affinity is removed
+            # from the blocks.
+            print "Disabling IP Pool"
+            pool.disabled = True
+            client.set_ip_pool_config(version, pool)
+
+            # Remove affinity from the allocation blocks for the pool.  This
+            # will prevent these blocks from being used for auto-allocations.
+            print "Removing IPAM configuration for pool"
+            client.release_pool_affinities(pool)
+            time.sleep(3)
+            client.release_pool_affinities(pool)
+
+            print "Deleting IP Pool"
```

Comment:
I think this is a bit much, just saying the pool is deleted below should be fine.

The `remove_ip_pool` command should be quick and there won't be a long period between this print and the one below, so the "Deleted" print will appear almost immediately after.